src/deterministic.jl

```diff
@@ -80,52 +144,55 @@ Construct Deterministic from a Dict of collections of data.
 """
 function Deterministic(
     name::AbstractString,
-    input_data::AbstractDict{Dates.DateTime, <:Any};
-    resolution::Dates.Period,
+    data::AbstractDict{Dates.DateTime, <:Any},
+    resolution::Dates.Period;
     normalization_factor::NormalizationFactor = 1.0,
     scaling_factor_multiplier::Union{Nothing, Function} = nothing,
 )
-    data = SortedDict{Dates.DateTime, Vector{Float64}}()
-    for (k, v) in input_data
-        try
-            data[k] = Float64[i for i in v]
-        catch e
-            @error("The forecast data provided $(second(eltype(input_data))) can't be converted to Vector{Float64}")
-            rethrow()
-        end
-    end
-    @assert !isempty(data)
-
-    return Deterministic(
+    return Deterministic(DeterministicStandard(
         name,
-        data;
-        resolution = resolution,
+        data,
+        resolution;
         normalization_factor = normalization_factor,
         scaling_factor_multiplier = scaling_factor_multiplier,
-    )
+    ))
+end
+
+"""
+Construct a Deterministic from a SingleTimeSeries.
+"""
+function Deterministic(ts::SingleTimeSeries, count, horizon, initial_timestamp, interval)
+    return Deterministic(DeterministicSingleTimeSeries(
+        single_time_series = ts,
+        count = count,
+        horizon = horizon,
+        initial_timestamp = initial_timestamp,
+        interval = interval,
+    ))
 end
 
 """
 Construct Deterministic from RawTimeSeries.
 """
 function Deterministic(
     name::AbstractString,
-    series_data::RawTimeSeries;
-    resolution::Dates.Period,
+    data::RawTimeSeries,
+    resolution::Dates.Period;
```

Comment:
I felt that it was odd to have two required arguments, one required keyword argument, and two optional keyword arguments.  I changed all of these cases to use three required arguments instead.